src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataSymbolMapExtensions.cs

```diff
@@ -1,28 +1,48 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Generic;
 using Microsoft.CodeAnalysis;
 
 namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
 {
     internal static class TaintedDataSymbolMapExtensions
     {
         /// <summary>
-        /// Determines if the given method is a tainted data source.
+        /// Determines if the given method is a potential tainted data source and get the related argument check methods.
         /// </summary>
         /// <param name="sourceSymbolMap"></param>
         /// <param name="method"></param>
+        /// <param name="evaluateWithPointsToAnalysis"></param>
+        /// <param name="evaluateWithValueContentAnslysis"></param>
         /// <returns></returns>
-        public static bool IsSourceMethod(this TaintedDataSymbolMap<SourceInfo> sourceSymbolMap, IMethodSymbol method)
+        public static bool IsSourceMethod(
+            this TaintedDataSymbolMap<SourceInfo> sourceSymbolMap,
+            IMethodSymbol method,
+            out HashSet<IsInvocationTaintedWithPointsToAnalysis> evaluateWithPointsToAnalysis,
+            out HashSet<IsInvocationTaintedWithValueContentAnalysis> evaluateWithValueContentAnslysis)
         {
+            evaluateWithPointsToAnalysis = new HashSet<IsInvocationTaintedWithPointsToAnalysis>();
+            evaluateWithValueContentAnslysis = new HashSet<IsInvocationTaintedWithValueContentAnalysis>(); ;
             foreach (SourceInfo sourceInfo in sourceSymbolMap.GetInfosForType(method.ContainingType))
```

Comment:
To avoid unnecessary allocations, I'd only instantiate the HashSets if they're needed (and of course the caller would have to check for null).

Reply:
Also consider using PooledHashSets.